File: skill-packs/flutter/flutter-fullstack-mobile/examples/good-example.dart
// GOOD — repository pattern, typed API client, error mapping
class OrderRepository {
  const OrderRepository(this._client);
  final ApiClient _client;

  Future<Order> createOrder(CreateOrderRequest request) async {
    final response = await _client.post('/orders', body: request.toJson());
    if (response.statusCode == 201) {
      return Order.fromJson(response.data);
    }
    throw ApiException.fromResponse(response);
  }
}

File: skill-packs/rust/axum/rust-axum-microservices/examples/good-example.rs
// GOOD — axum handler with State, typed Json, error mapping
async fn create_order(
    State(state): State<AppState>,
    Json(payload): Json<CreateOrderRequest>,
) -> Result<(StatusCode, Json<OrderResponse>), AppError> {
    let order = state.orders.create(payload).await?;
    Ok((StatusCode::CREATED, Json(order.into())))
}
